test/stepper.js

```diff
@@ -62,6 +62,349 @@ exports["Stepper Firmware Requirement"] = {
+      stepper.pins,
+      {motor1: pins[0], motor2: pins[1], motor3: pins[2], motor4: pins[3]}
+    );
+
+    test.done();
+  },
+};
+
+exports["Stepper - max steppers"] = {
+  setUp: function(done) {
+
+    this.maxSteppers = 6;
+
+    this.board1 = new Board({
+      io: new MockFirmata({
+        pins: [
+          {
+            supportedModes: [8],
+          },
+        ]
+      }),
+      debug: false,
+      repl: false
+    });
+
+    this.board2 = new Board({
+      io: new MockFirmata({
+        pins: [
+          {
+            supportedModes: [8],
+          },
+        ]
+      }),
+      debug: false,
+      repl: false
+    });
+
+    done();
+  },
+
+  tearDown: function(done) {
+    done();
+  },
+
+  singleBoard: function(test) {
+    test.expect(2);
+
+    test.doesNotThrow(function() {
+      for (var i = 0; i < this.maxSteppers; i++) {
+        new Stepper({
+          board: this.board1,
+          stepsPerRev: 200,
+          pins: [3, 4, 5, 6]
+        });
+      }
+    }.bind(this));
+
+    test.throws(function() {
+      new Stepper({
+        board: this.board1,
+        stepsPerRev: 200,
+        pins: [3, 4, 5, 6]
+      });
+    }.bind(this));
+
+    test.done();
+  },
+
+  multipleBoards: function(test) {
+    test.expect(3);
+
+    // should be able to add max on two boards
+    test.doesNotThrow(function() {
+      for (var i = 0; i < this.maxSteppers; i++) {
+        new Stepper({
+          board: this.board1,
+          stepsPerRev: 200,
+          pins: [2, 3, 4, 5]
+        });
+        new Stepper({
+          board: this.board2,
+          stepsPerRev: 200,
+          pins: [2, 3, 4, 5]
+        });
+      }
+    }.bind(this));
+
+    // but can't add any more to either board
+    test.throws(function() {
+      new Stepper({
+        board: this.board1,
+        stepsPerRev: 200,
+        pins: [2, 3, 4, 5]
+      });
+    }.bind(this));
+    test.throws(function() {
+      new Stepper({
+        board: this.board2,
+        stepsPerRev: 200,
+        pins: [2, 3, 4, 5]
+      });
+    }.bind(this));
+
+    test.done();
+  }
+};
+
+// TODO test that callback is called for .step()
+// mock firmata apparently never calls it, so we can't test currently
+// exports["Stepper - step callback"] = {};
+
+exports["Stepper - set direction required before step"] = {
+  setUp: function(done) {
+
+    this.board = new Board({
+      io: new MockFirmata({
+        pins: [
+          {
+            supportedModes: [8],
+          },
+        ]
+      }),
+      debug: false,
+      repl: false
+    });
+
+    this.stepper = new Stepper({
+      board: this.board,
+      type: five.Stepper.TYPE.DRIVER,
+      stepsPerRev: 200,
+      pins: [2, 3]
+    });
+    done();
+  },
+
+  tearDown: function(done) {
+    done();
+  },
+
+  directionSet: function(test) {
+    test.expect(1);
+
+    // If we haven't set a direction, we should get error
+    this.stepper.step(1, function(error) {
+      test.ok(!!error);
```

Comment:
Right now, testing the error message is the best option if you don't want to change the code that's throwing the error. Since error message should never be localized, it's actually a fairly safe option. Any change to the error message would require a change to the tests, but those should be trivial.

A common pattern in node is to assign a `code` property which contains a string which can be used similar to a constant. For example, `"EISDIR"` and `"ENOENT"` are common error codes from the `fs` module. This is generally nicer to work with in JS rather than custom `Error` subclasses. This can make the code a bit verbose since you can't just write `callback(new Error("..."))`; I've dealt with that before by creating a utility method: https://github.com/scottgonzalez/debt/blob/13d43f3b116df7e769cefaf58f5d8d0e475c64c2/lib/util.js#L100